packages/flutter_tools/test/data/dart_dependencies_test/syntax_error/foo.dart

```diff
@@ -2,4 +2,4 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-bad programmer!
+import bad programmer!
```

Comment:
With this change, we only parse the directive section of dart sources to figure out the dependencies.  This syntax error was technically after the directive section and is therefore ignored by the new DartDependencySetBuilder. Introducing the import statement makes the syntax error part of the directive section.

Now you might ask: "But what about syntax errors after the directive section during hot reload?" After the changed sources are pushed to device they are parsed (again) by the Dart VM on device, which will provide a clear and actionable error message if it encounters any syntax errors. So, everything is good here as well.